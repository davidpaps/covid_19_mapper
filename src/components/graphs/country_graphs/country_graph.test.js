import * as React from "react";
import { shallow } from "enzyme";
import CountryGraphContainer from "./country_graph";
import CountryLineData from "./country_data_line/country_data_line";
import GrowthFactorLine from "./growth_factor_line/growth_factor_line";
import ComparisonLineContainer from "./comparison_line/comparison_line";
import DailyChangesBar from "./daily_changes_bar/daily_changes_bar";
import CFRContainer from "./country_cfr/country_cfr";

describe("Country Graph Container", () => {
  let wrapper;

  beforeEach(
    () =>
      (wrapper = shallow(
        <CountryGraphContainer
          country={" "}
          countries={[]}
          total={[]}
          timeseries={[]}
        />
      ))
  );

  it("should render a <div />", () => {
    expect(wrapper.find("div").length).toEqual(6);
  });

  it("should render the CountryLineData Component", () => {
    expect(wrapper.containsMatchingElement(<CountryLineData />)).toEqual(true);
  });

  it("should render the GrowthFactorLine Component", () => {
    expect(wrapper.containsMatchingElement(<GrowthFactorLine />)).toEqual(true);
  });

  it("should render the ComparisonLineContainer Component", () => {
    expect(
      wrapper.containsMatchingElement(<ComparisonLineContainer />)
    ).toEqual(true);
  });

  it("should render the DailyChangesBar Component", () => {
    expect(wrapper.containsMatchingElement(<DailyChangesBar />)).toEqual(true);
  });

  it("should render the CFRContainer Component", () => {
    expect(wrapper.containsMatchingElement(<CFRContainer />)).toEqual(true);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
