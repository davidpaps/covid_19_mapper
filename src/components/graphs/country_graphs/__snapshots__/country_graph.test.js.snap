// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Graph Container renders correctly 1`] = `ShallowWrapper {}`;
